File: aspnet-core/src/Lesson.Application/Domain/Homework/Dto/CreateHomeworkInput.cs
using Microsoft.AspNetCore.Http;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lesson.Domain.Homework.Dto
{
    public class CreateHomeworkInput
    {
        public string Summary { get; set; }
        public string Description { get; set; }
        public long TeacherId { get; set; }
        public int ClassRoomId{ get; set; }
        public int LessonId { get; set; }
        public List<IFormFile> Files { get; set; }

    }
}

File: angular/src/app/services/Homework/dtos/CreateHomeworkInput.ts
export class CreateHomeworkInput implements ICreateHomeworkInput{
    summary: string="";
    description: string="";
    teacherId: number;
    classRoomId: number;
    lessonId: number;
    files: FormData;
    constructor(data?:ICreateHomeworkInput){}
}

export interface ICreateHomeworkInput{
    summary:string|undefined;
    description:string|undefined;
    teacherId:number|undefined;
    classRoomId:number|undefined;
    lessonId:number|undefined;
    files:FormData|undefined;
}
